app/poros/search.rb

```diff
@@ -0,0 +1,26 @@
+class Search
+  def initialize(search_param)
+    @search_param = search_param
+  end
+
+  def from_product
+    Product.where("lower(name) like ?", "%#{@search_param.downcase}%")
+  end
+
+  def from_brand
+    Product.where("lower(brand) like ?", "%#{@search_param.downcase}%")
+  end
+
+  def from_category
+    subcategories = Subcategory.where("lower(name) like ?", "%#{@search_param.downcase}%")
+    products = Product.where("lower(name) = ?", "")
+    subcategories.each do |subcategory|
+      products << subcategory.products
+    end
+    byebug
```

Comment:
Sorry about that. Our first implementation of the search feature used a poros, which was later removed.